### PR TITLE
Добро пожаловать, или посторонним вход воспрещён (часть 1)

### PR DESCRIPTION
**Pull request by olegsorokin**
---
:mortar_board: [Добро пожаловать, или посторонним вход воспрещён (часть 1)](https://up.htmlacademy.ru/react/9/user/30818/tasks/11)

### CODE REVIEW

File: project/src/store/action.ts [hunk 1]
```diff
@@ -2,11 +2,14 @@ import { createAction } from '@reduxjs/toolkit';
...
 
 const CHANGE_GENRE = 'CHANGE_GENRE';
 const LOAD_FILMS = 'LOAD_FILMS';
 const LOAD_PROMO = 'LOAD_PROMO';
+const REQUIRE_AUTHORIZATION = 'REQUIRE_AUTHORIZATION';
 
 export const changeGenre = createAction<GenreEnum>(CHANGE_GENRE);
 export const loadFilms = createAction<Film[]>(LOAD_FILMS);
 export const loadPromo = createAction<Film>(LOAD_PROMO);
+export const requireAuthorization = createAction<AuthorizationStatus>(REQUIRE_AUTHORIZATION);
```
Comment (dianatsevan) [inline project/src/store/action.ts:15]: Probably, you wight want to rename the var to `requestAuthorization`. It seems it `request` sounds better

File: project/src/types/user-data.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export type UserData = {
```
Comment (dianatsevan) [inline project/src/types/user-data.ts:1]: I think we don't need to use `data` word in the of the type and file (auth-data - the same)

File: project/src/services/error-handle.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import request from 'axios';
+import { toast } from 'react-toastify';
+
+import { HttpCode } from '../constants/http';
+import { TError } from '../types/error';
+
+export function errorHandle(error: TError): void {
```
Comment (dianatsevan) [inline project/src/services/error-handle.ts:7]: Is the name `errorHandle` from the example? `handleError` ?

File: project/src/services/error-handle.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  if (response) {
+    switch (response.status) {
+      case HttpCode.BAD_REQUEST:
+        toast.info(response.data.error);
+        break;
+      case HttpCode.UNAUTHORIZED:
+        toast.info(response.data.error);
+        break;
+      case HttpCode.NOT_FOUND:
+        toast.info(response.data.error);
```
Comment (dianatsevan) [inline project/src/services/error-handle.ts:23]: `toast.info(response.data.error);` is the same for each case. Don't you think we could enhance the switch?

File: project/src/pages/sign-in/sign-in.tsx [hunk 5]
```diff
@@ -1,6 +1,37 @@
...
+import { useAppDispatch } from '../../hooks';
+import { login } from '../../store/api-actions';
+import { AppRoute } from '../../constants/routs';
 
 function SignIn(): JSX.Element {
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const [formData, setFormData] = useState({
+    userEmail: '',
```
Comment (dianatsevan) [inline project/src/pages/sign-in/sign-in.tsx:14]: I think `user` word usage is redundant here

**Review by dianatsevan — APPROVED** [review]

File: project/src/store/action.ts (hunk 1 above)
Comment (olegsorokin) [reply project/src/store/action.ts:15]: This request is to check the authorization status. Will not authorize the user.

File: project/src/services/error-handle.ts (hunk 3 above)
Comment (olegsorokin) [reply project/src/services/error-handle.ts:7]: Yes, from an example. I think, we can change that.

**Pull request merged**